analyzer/txtracer.go

```diff
@@ -113,6 +113,23 @@ func (tr *Tracer) TraceTransaction(blockHeader *types.Header, tx *types.Transact
 		}
 
 		ops = append(ops, reconciledOps...)
+	} else if receipt.Status == types.ReceiptStatusFailed {
+		contractMap, err := tr.GetRegistryAddresses(receipt, registry.ReserveContractID.String(), registry.LockedGoldContractID.String(), registry.ElectionContractID.String(), registry.GovernanceContractID.String(), registry.AccountsContractID.String())
+		if err != nil {
+			return nil, err
+		}
+
+		tobinTax, err := tr.GetTobinTax(receipt.BlockNumber, contractMap[registry.ReserveContractID.String()])
+		if err != nil {
+			return nil, err
+		}
+
+		transferOps, err := tr.TxTransfers(tx, receipt, tobinTax)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Hi @shrimalmadhur although this will produce operations with failed statuses the output would be inconsistent with the output for a successful transaction because `ReconcileLogOpsWithTransfers` is not being called.

Below is a diff of the `block/transaction` output for a successful transaction did not undergo reconciliation and the same transaction with reconciliation. We can see that reconciliation changes the type of some operations and also introduces additional operations. So if you only want to use these operations to indicate failure then that should be ok, but if you try to extract any further meaning from these failed operations I think it is likely to cause problems for you.

```diff
--- a/locktx-no-reconcile.json
+++ b/locktx-with-reconcile.json
@@ -80,7 +80,7 @@
                "index" : 3
             },
             "status" : "success",
-            "type" : "transfer"
+            "type" : "lockGold"
          },
          {
             "account" : {
@@ -102,7 +102,35 @@
                }
             ],
             "status" : "success",
-            "type" : "transfer"
+            "type" : "lockGold"
+         },
+         {
+            "account" : {
+               "address" : "0xF67252F48D986CE971306242C4dc24B54399425D",
+               "sub_account" : {
+                  "address" : "LockedGoldNonVoting"
+               }
+            },
+            "amount" : {
+               "currency" : {
+                  "decimals" : 18,
+                  "symbol" : "cGLD"
+               },
+               "value" : "1794751700000000"
+            },
+            "operation_identifier" : {
+               "index" : 5
+            },
+            "related_operations" : [
+               {
+                  "index" : 3
+               },
+               {
+                  "index" : 4
+               }
+            ],
+            "status" : "success",
+            "type" : "lockGold"
          }
       ],
       "transaction_identifier" : {
```

Reply:
@piersy quick questions:
1. If we add `ReconcileLogOpsWithTransfers`, does the amount show 0 for this failed transaction?
2. Does all failed transaction in Celo only charge fee? In other words, will a failed txn has balance change except fee?

Reply:
Hi @xiaying-peng the above diff is for a successful transaction.
1 . If you add `ReconcileLogOpsWithTransfers` for a failed transaction then you will only see fee the fee paying operations for the failed transaction.
2. Failed transactions will only pay fees, no other balance change will occur.